dotnet/samples/Concepts/FunctionCalling/FunctionCalling.cs

```diff
@@ -0,0 +1,459 @@
+    }
+
+    [Fact]
+    /// <summary>
+    /// This example demonstrates usage of the non-streaming chat completion API with <see cref="FunctionChoiceBehavior.Auto"/> that advertises all kernel functions to the AI model and invokes them automatically.
+    /// </summary>
+    public async Task RunNonStreamingChatCompletionApiWithAutomaticFunctionInvocationAsync()
+    {
+        Kernel kernel = CreateKernel();
+
+        // To enable automatic function invocation, set the `autoInvoke` parameter to `true` in the line below or omit it as it is `true` by default.
+        OpenAIPromptExecutionSettings settings = new() { FunctionChoiceBehavior = FunctionChoiceBehavior.Auto() };
+
+        IChatCompletionService chatCompletionService = kernel.GetRequiredService<IChatCompletionService>();
+
+        ChatMessageContent result = await chatCompletionService.GetChatMessageContentAsync(
+            "Given the current time of day and weather, what is the likely color of the sky in Boston?",
+            settings,
+            kernel);
+
+        // Assert
+        Console.WriteLine(result);
+
+        // Expected output: "The likely color of the sky in Boston is gray due to the current rainy weather."
+    }
+
+    [Fact]
+    /// <summary>
+    /// This example demonstrates the usage of the streaming chat completion API with <see cref="FunctionChoiceBehavior.Auto"/> that advertises all kernel functions to the AI model and invokes them automatically.
+    /// </summary>
+    public async Task RunStreamingChatCompletionApiWithAutomaticFunctionInvocationAsync()
+    {
+        Kernel kernel = CreateKernel();
+
+        // To enable automatic function invocation, set the `autoInvoke` parameter to `true` in the line below or omit it as it is `true` by default.
+        OpenAIPromptExecutionSettings settings = new() { FunctionChoiceBehavior = FunctionChoiceBehavior.Auto() };
+
+        IChatCompletionService chatCompletionService = kernel.GetRequiredService<IChatCompletionService>();
+
+        var stringBuilder = new StringBuilder();
+
+        // Act
+        await foreach (var update in chatCompletionService.GetStreamingChatMessageContentsAsync(
+            "Given the current time of day and weather, what is the likely color of the sky in Boston?",
+            settings,
+            kernel))
+        {
+            stringBuilder.Append(update.Content);
+        }
+
+        // Assert
+        Console.WriteLine(stringBuilder.ToString());
+
+        // Expected output: "Given that it's currently daytime and rainy in Boston, the sky is likely to be grey or overcast."
+    }
+
+    /// <summary>
+    /// This example demonstrates the usage of the non-streaming chat completion API with <see cref="FunctionChoiceBehavior.Auto"/> that advertises all kernel functions to the AI model and invokes them manually.
+    /// </summary>
+    [Fact]
+    public async Task RunNonStreamingChatAPIWithManualFunctionInvocationAsync()
```

Comment:
nit: To align with other method names :) 
```suggestion
    public async Task RunNonStreamingChatCompletionApiWithManualFunctionInvocationAsync()
```